src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -123,6 +126,11 @@ public virtual IVoidConfiguration Invokes(Action<IFakeObjectCall> action)
 
         public virtual IAfterCallConfiguredConfiguration<IVoidConfiguration> CallsBaseMethod()
         {
+            if (this.manager.FakeObjectType.GetTypeInfo().IsSubclassOf(typeof(Delegate)))
+            {
+                throw new FakeConfigurationException(ExceptionMessages.DelegateCannotCallBaseMethod);
+            }

```

Comment:
Is there a reason you chose this location for the check? I would've picked the `BuildableCallRule.CallBaseMethod` setter. Both classes are internal, and the property isn't implementing any publicly visible member, so I think it makes no difference, but I'm curious.

Reply:
It seemed the most natural place at the time. Why do you think the property setter would be better?

Reply:
I'm not sure I do think it! I'm genuinely just asking a question. I started from the existing exception and walked up the stack. I suspect you started at the test's call site and walked down, and we just arrived at different places. I guess it's possible we later add a new rule that allows base-calling, and it bypasses your check, but I really doubt it's going to happen. The current location is fine. It was the wording of the exception that I thought would have a bigger impact.

Reply:
Also, in `BuildableCallRule` we don't know the type of the fake...